### DISCUSSION TITLE
특정 구역에 가면 캐릭터 Teleport하기

### DISCUSSION THREAD

**Discussion by smartbh** [naverz/zepeto-studio-kor#18] [5. Q&A]
건물의 입구같은곳을 가면 캐릭터를

건물의 내부로 꾸며진 다른 장소 (다른 씬x)로

텔레포트 시키고 싶습니다.

하지만 캐릭터스폰하는 emptyGameObject에 임포트 해놓은 스크립트의 캐릭터 플레이어를

다른 빈 collider만 가진 empty Game Object의 함수에서 OnTriggerEnter로 태그를 확인, 플레이어가 맞으면

텔레포트를 시키고 싶습니다. 반대의 방향도 마찬가지구요 (건물 내부의 문 -> 바깥 맵으로 텔레포트)

이런 경우의 예제를 알고싶습니다.

현재 몇가지 코드를 짜서 시도해 봤는데 오류가 좀 있어서 질문을 남기게 됐습니다.

![image](https://user-images.githubusercontent.com/43138813/144103332-230d6286-0750-4632-9940-f16aa45b5675.png)

텔레포트 스크립트에서 제페토 플레이어를 만들어서 태그나 이름으로 시도해 봤는데

![image](https://user-images.githubusercontent.com/43138813/144103366-511c38a1-03d3-49e9-b1ac-027b61ee9f02.png)

타입에러에 프로퍼티를 읽지 못하는 오류가 나왔습니다.

이 경우 제대로 내용물을 불러오지 못해 빈 데이터를 OnTrigger함수에서 읽고

Tag나 Name을 찾아서 비교를 해보려다가 발생하는 오류같은데

해결방법을 알고싶습니다.

현재 Local플레이어 버전으로 연습중이고 추후 Multiplay 플레이어 버전으로 바꿀 생각입니다.

**Comment by moondory77-naverz** [comment]
안녕하세요. 
OnTriggerEnter(collider)로 들어오는 이벤트의 경우,
전달되는 collider 객체의 type은 UnityEngine.Collider 입니다.

이를 ZepetoPlayer 타입 변수에 할당해서 발생하는 문제 같습니다.

참고로 제페토 스크립트의 경우, Unity의 각종 Event를 동일한 형태로 구현하고 있습니다. 
또 문의가 있으실 경우, 언제든 말씀 부탁드리겠습니다. 감사합니다.
